### ISSUE TITLE
Attempting to refer to class component in template literal causes rendering the parent with enzyme's mount to throw

### ISSUE THREAD

**Issue by dgcoffman** [styled-components/styled-components#1314]
Attempting to refer to class component in template literal as described at https://www.styled-components.com/docs/faqs#can-i-refer-to-other-components causes rendering the parent with enzyme's mount to throw.

## Version
"styled-components": "2.2.1",
"jest": "21.0.2",
"babel-jest": "21.0.2",
"jest-styled-components": "4.6.0",
"babel-plugin-styled-components": "1.1.7",
"enzyme": "3.1.0",

## Reproduction

```
import React from 'react';
import styled from 'styled-components';
import { mount } from 'enzyme';

class A extends React.Component<{}> {
  render() {
    return <div />;
  }
}

const B = styled.div`
  ${A} {
  }
`;

test('B throws TypeError: Cannot call a class as a function', () => {
  mount(<B />);
});
```

## Steps to reproduce
Run the unit test with Jest.

## Expected Behavior
The component renders.

## Actual Behavior
```
 FAIL  src/components/Text/Text-test.js
  ✕ B throws TypeError: Cannot call a class as a function (16ms)

  ● B throws TypeError: Cannot call a class as a function

    TypeError: Cannot call a class as a function

      at _classCallCheck (src/components/Text/Text-test.js:4:222)
      at A (src/components/Text/Text-test.js:6:79)
      at node_modules/styled-components/lib/utils/flatten.js:42:72
          at Array.reduce (<anonymous>)
      at flatten (node_modules/styled-components/lib/utils/flatten.js:30:17)
      at ComponentStyle.generateAndInjectStyles (node_modules/styled-components/lib/models/ComponentStyle.js:80:21)
      at StyledComponent.generateAndInjectStyles (node_modules/styled-components/lib/models/StyledComponent.js:135:40)
      at StyledComponent.componentWillMount (node_modules/styled-components/lib/models/StyledComponent.js:178:40)
      at callComponentWillMount (node_modules/react-dom/cjs/react-dom.development.js:9777:14)
```

**Comment by philpl** [comment]
First of all, thank you for following the issue template and being so detailed! Hats off 👍🙂

The component A you’re referring to is not a styled component. The class name interpolation works because we have consistent “component ids” for all styled components.

We also use these ids to attach a static, empty class name. You might’ve noticed it in the devtools.

Now when you use the interpolation with a styled component we just take that class and insert it into your CSS string, thus this doesn’t work for any regular components

I’ve noticed that the docs aren’t that clear on this and a PR is certainly welcome, except of course if @oliverturner, the author of this section, would like to fix this.

> which lets **components** encapsulate the entirety of their styling: as with media queries, it lets **components** describe how they will behave when affected by external changes

So to recap, when that sections says components, it is referring to styled components

**Comment by oliverturner** [comment]
Good point there: will PR for clarity

________________________________
